src/AngleSharpHttpResponseMessageResponse.cs

```diff
@@ -0,0 +1,75 @@
+﻿using AngleSharp;
+using AngleSharp.Common;
+using AngleSharp.Io;
+
+using Flurl.Http;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace PlayerIO
+{
+    // Rather than having some ugly OpenAsync with .Results to convert Task<T>s to Ts, having his class allows us to
+    // execute the exact method of AngleSharp which takes an IResponse. This prevents that .Result hell, and allows
+    // for more truly asynchronous code.
+    //
+    // The primary usecase of this is in FlurlClientExtensions.LoadDocumentAsync(IFlurlRequest, CancellationToken),
+    // which utilizes the AngleSharpHttpResponseMessageResponseExtensions.ToAngleSharp method.
+
+    // yes the name is ugly, idk what to name it
+    internal class AngleSharpHttpResponseMessageResponse : IResponse
```

Comment:
I think 'ASMessageResponse' could be a good name?

Reply:
It conveys exactly what it is, as is, and I don't like the idea of taking meaning out of the name - but it's internal and nobody cares, so sure.